ShakeToCharge.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect } from 'react';
+import { View, Text, StyleSheet, Vibration, Button } from 'react-native';
+import { Accelerometer } from 'expo-sensors';
+import * as Battery from 'expo-battery';
+
+const ShakeToCharge = () => {
+  const [batteryLevel, setBatteryLevel] = useState<number>(-1);
+  const [batteryColor, setBatteryColor] = useState<string>('green');
```

Comment:
`batteryColor` can be a regular variable since it's value is derived from `batteryLevel` state. So anytime the level, changes, we cause a re-render which also updates `batteryColor`